examples/swarmmode-rhel.json

```diff
@@ -0,0 +1,37 @@
+{
+  "apiVersion": "vlabs",
+  "properties": {
+    "orchestratorProfile": {
+      "orchestratorType": "SwarmMode"
+    },
+    "masterProfile": {
+      "count": 3,
+      "dnsPrefix": "",
+      "vmSize": "Standard_D2_v2"
+    },
+    "agentPoolProfiles": [
+      {
+        "name": "agentpublic",
+        "count": 3,
+        "vmSize": "Standard_D2_v2",
+        "dnsPrefix": "",
+        "ports": [
+          80,
+          443,
+          8080
+        ]
+      }
+    ],
+    "linuxProfile": {
+      "adminUsername": "azureuser",
+      "ssh": {
+        "publicKeys": [
+          {
+            "keyData": ""
+          }
+        ]
+      },
+      "distro": "rhel"

```

Comment:
I reviewed this with @anhowe, and we agreed that the `distro` property should be a agent pool profile property (in the `agentPoolProfiles` array). The reason for this is that we want to support different distributions for different agent pools (there may be use cases where a heterogeneous cluster makes sense: _n_ nodes using distro ABC, and _n_ nodes using distro XYZ).

We also want the master profile to have this option as well.

Thank you for all this work, we really *do* want this feature! (We just want to be able to re-use it, and I understand your use-case is swarm mode-specific; so apologies if we're asking you to go the extra mile!)

Reply:
Thanks Jack, that's the kind of feedback I was looking for :) Indeed my partner's requirement is to run all nodes on RHEL because most of their customers have standardized on that distribution exclusively. They don't have a requirement for heterogeneous clusters. I will evaluate the work needed to move the distro parameter at the master/agent level and get back to you.

Reply:
Since we are going to support multiple distros per deployment, and the generated template can contain bits specific to one distro (e.g. cloud-init vs CSE), I will need to remove the OS specs from the deployment parameters and move them to template variables. There will be one set for the master, and then one set per agent pool. Is that all right with you guys?